core/sql/optimizer/BindRelExpr.cpp

```diff
@@ -1573,7 +1573,8 @@ NATable *BindWA::getNATable(CorrName& corrName,
   // native HIVE or HBASE objects unless the allowExternalTables flag is set.  
   // allowExternalTables is set for drop table and SHOWDDL statements.  
   // TDB - may want to merge the Trafodion version with the native version.
-  if ((table) && (table->isExternalTable() && (! bindWA->allowExternalTables())))
+  if ((table) && (table->isExternalTable() && (! bindWA->allowExternalTables()) &&
```

Comment:
As an option, you could set "allowExternalTables" in the code that get the NATable structure instead of adding another check.  Not sure how many places in Histogram code you need to get NATable information.